### PR TITLE
Feature/blockchain introduction

### PR DESCRIPTION
**Pull request by kencoenen**

### CODE REVIEW

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+
+> A lot of people are talking about blockchain these days.
+They're talking about blockchain as the next big thing after mainframes, computers, the internet and social networking.
+This introduction is the first part in a series of blockchain posts.
+
+# Topics
+
+In this first article about the innovative blockchain technology, we'll cover the following topics:
+
+1. [Blockchain and it's relation to Bitcoin](#blockchain-and-its-relation-to-bitcoin)
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: its

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+4. [The consensus process](#the-consensus-process)
+5. [Smart contracts](#smart-contracts)
+6. [Valid blockchain business cases](#valid-blockchain-business-cases)
+7. [Existing platforms](#existing-platforms)
+8. [Thinking decentralized](#thinking-decentralized)
+9. [Conclusion](#conclusion)
+10. [Recommended reading](#recommended-reading)
+
+
+# Blockchain and it's relation to Bitcoin
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: its

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+  <div class="col-md-2" style="width:32%">
+	{% include image.html img="/img/blockchain/bitcoin_blockchain.png" alt="Blockchain" title="Blockchain" caption="3. The blockchain which stores all Bitcoin transactions" style="max-width: 80px;" %}
+</div>
+</div>
+
+So when talking about Bitcoin or blockchain with people, it's important to mind this terminology.
+Here's a funny quote I read in the book [Blockchain: A Blueprint for a New Economy](http://www.goodreads.com/book/show/24714901-blockchain),
+which describes this ambiguity very well:
+<blockquote class="clear"><p>
+It's as if PayPal called the internet PayPal on which the PayPal protocol was run to transfer PayPal currency.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: "was run" sounds odd...
"Is running"?

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+</p></blockquote>
+
+In January 2009, the bitcoin network came into existence.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Bitcoin (capital 'b')

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+Bitcoin isn't the first attempt to digital currency, but it's the first that uses a __peer-to-peer network__ to create a platform for executing transactions __without depending on central authorities__ who validate them.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: first **one**?

Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: who validate**s** them

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 6]
```diff
@@ -0,0 +1,422 @@
...
+
+You should see Bitcoin as the first platform that implemented blockchain technology.
+
+
+# What is blockchain?
+
+So forget about Bitcoin now.
+That's not what this post is about.
+People say blockchain is as important as the introduction of the internet. 
+The internet is a world wide network to __share information__ with one and another, but it is far less suitable for transferring value.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Should be either "world-wide" or "worldwide".

Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: I think its "one another" instead of "one and another".

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 7]
```diff
@@ -0,0 +1,422 @@
...
+A miner builds a block containing all transactions, a proof of work that the puzzle was solved (also known as the __block root hash__, which is also the ID of the block) and a hash to the previous block.
+
+A block also contains the following items:
+
+* A timestamp
+* a nonce
+* and a __merkle root hash__
+
+A __merkle root__ does not verify transactions, but verifies a set of transactions.
+Transaction ID's are hashes of the transaction, and the merkle tree is constructed from these hashes.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Transaction ID**s**

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 8]
```diff
@@ -0,0 +1,422 @@
...
+It means that if a single detail in any of the transactions changes, so does the merkle root. 
+It also means that if the exact same transactions are listed in a different order, the merkle root will also change.
+So the merkle root is cryptographic proof of which transactions are in the block, and which order they are in.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Would rephrase:
So the merkle root is cryptographic proof of **the transactions in the block together with the** order they are in.

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 9]
```diff
@@ -0,0 +1,422 @@
...
+
+The __nonce number__ specifies that the block root hash must start with a specified number, like 4 leading zero’s.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Four instead of 4.

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 10]
```diff
@@ -0,0 +1,422 @@
...
+This way blockchain is very secure.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: This way**,** ... (comma)

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 11]
```diff
@@ -0,0 +1,422 @@
...
+It also means that if the exact same transactions are listed in a different order, the merkle root will also change.
+So the merkle root is cryptographic proof of which transactions are in the block, and which order they are in.
+
+The __nonce number__ specifies that the block root hash must start with a specified number, like 4 leading zero’s.
+This way blockchain is very secure. 
+Sha-256 is used to hash.
+
+The miner appends the block to the blockchain. 
+And the majority of the other nodes, 50% + 1, double-check by __verifying the proof of work__ in the block.
+It sometimes occurs that miners will validate 2 blocks at the same time and they will be appended to the chain.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Two instead of 2

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 12]
```diff
@@ -0,0 +1,422 @@
...
+You can read more about [the consensus process](#the-consensus-process) further in this blogpost.
+
+The public blockchain is generally a complete peer-to-peer network. Its characteristics are:
+
+* The users from the chain get protected from the creators of the chain, because there are actions to the network that even they cannot perform.
+__Developers are not the owners__ of the network and don't have more or less privileges than normal users.
+* These chains are __transparent__ because everyone can see what is happening inside the chain.
+
+<blockquote class="clear"><p>
+“In some cases, public is clearly better; in others, some degree of private control is simply necessary.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Probably best to swap the Word quotes by actual quotes.
“ -> "

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 13]
```diff
@@ -0,0 +1,422 @@
...
+ As is often the case in the real world, it depends.”
+ - Vitalik Buterin of Ethereum
+</p></blockquote>
+
+
+### Consortium blockchains
+
+In this type of blockchain network, the consensus process is executed by a __predetermined group of nodes__ in the network.
+Let's take a consortium of 15 financial institutions as an example, each with a node.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Fifteen instead of 15

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 14]
```diff
@@ -0,0 +1,422 @@
...
+From this group of 15, there are 10 nodes that need to sign each block before it is valid.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Fifteen instead of 15
Ten instead of 10

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 15]
```diff
@@ -0,0 +1,422 @@
...
+You could say that these 10 take ownership of the data in the blockchain.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Ten

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 16]
```diff
@@ -0,0 +1,422 @@
...
+They decide which transactions are valid and which are not.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: ... and which **ones** are not.

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 17]
```diff
@@ -0,0 +1,422 @@
...
+So what is it?  
+Smart contracts are traditional contracts and official documents, but written in code.
+As such, the contract is understandable for everyone across the globe, irrespective of the jurisdiction it is related to. 
+Smart contracts are like **If This Then That** statements, only they tend to be a lot more complex.
+
+The different definitions usually fall into one of the following two categories:
+
+1. Sometimes the term is used to identify a specific technology.
+__Pieces of code__ that are stored, verified and executed on a blockchain.
+For example a hello world program.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: For example**,** a ....

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 18]
```diff
@@ -0,0 +1,422 @@
...
+When we translate the blockchain characteristics to business values, it can potentially solve business problems in the following five key elements:
+
+1. [Transparency](#transparency)
+2. [Operation harmonization](#operation-harmonization)
+3. [Business continuity](#business-continuity)
+4. [Permanence](#permanence)
+5. [Security](#security)
+6. [Decentralized](#thinking-decentralized)
+
+We'll discuss each element in more detail and explain why blockchain technology can be an answer to that business problem.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Think you can just say "in detail" instead of "in more detail".

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 19]
```diff
@@ -0,0 +1,422 @@
...
+
+
+### Transparency
+
+In a public blockchain network, by default every member of the ecosystem can access all transactions stored in the chain.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: by default**,** ... (comma)

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 20]
```diff
@@ -0,0 +1,422 @@
...
+It could be interesting to combine both blockchain technology and Event Sourcing principles in a project.
+
+If you want to learn more about Event Sourcing, make sure to visit the following pages:
+
+* There's an excellent article on [Martin Fowler's blog](http://martinfowler.com/eaaDev/EventSourcing.html)
+* Our colleague Yannick De Turck also has a chapter on Event Sourcing in his [blogpost about Lagom](/microservices/2016/04/22/Lagom-First-Impressions-and-Initial-Comparison-to-Spring-Cloud.html#cqrs-and-event-sourcing)
+* Ken Coenen has written about [CQRS and Event Sourcing](/domain-driven%20design/2016/02/02/A-Decade-Of-DDD-CQRS-And-Event-Sourcing.html) too after Ordina JWorks was present at [DDD Europe](https://dddeurope.com) back in 2016
+
+> Please note that you can only achieve __full immutability__ if you setup a public, permissionless blockchain network.
+In a consortium- or private blockchain network, transactions can be altered because you know the nodes that validate them.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Think you can leave out the dash (-).

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 21]
```diff
@@ -0,0 +1,422 @@
...
+### Security
+
+Blocks are timestamped and protected with __cryptographic technology__ that is considered unbreakable.
+If a block is added it can't be removed or altered.
+If you change a single bit of a transaction, the hash of this transaction will be completely different. 
+So the __merkle root__ hash (Merkle trees are explained in the section [What is blockchain](#what-is-blockchain)) won't be the same, the __nonce__ number will then be wrong and the block will be considered invalid.
+In this way transactions are secure once chained to the blockchain. 
+
+The cryptographic technology works with the principal of __public and private keys__, but hashing is also a part of this technology.
+The private key is linked to the public key, but you can not find out the private key if you have the public key.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Although also correct, I think **cannot** is preferred. :-)

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 22]
```diff
@@ -0,0 +1,422 @@
...
+Blocks are timestamped and protected with __cryptographic technology__ that is considered unbreakable.
+If a block is added it can't be removed or altered.
+If you change a single bit of a transaction, the hash of this transaction will be completely different. 
+So the __merkle root__ hash (Merkle trees are explained in the section [What is blockchain](#what-is-blockchain)) won't be the same, the __nonce__ number will then be wrong and the block will be considered invalid.
+In this way transactions are secure once chained to the blockchain. 
+
+The cryptographic technology works with the principal of __public and private keys__, but hashing is also a part of this technology.
+The private key is linked to the public key, but you can not find out the private key if you have the public key. 
+The private key allows you to verify that you are the owner of the public key. 
+To make transactions, you'll need a unique key (private key) to make a digital signature to prove you are the owner.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: ... to prove **that** you ...

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 23]
```diff
@@ -0,0 +1,422 @@
...
+The proof-of-work can also help deter these attacks and spam because of the high costs of mining. 
+Even if a hacker is able to penetrate one network and attempts to steal funds, there are multiple redundant copies of the same ledger stored around the world. 
+If one is tampered with, the others could be used as a backup to prove what funds actually belong in each account. 
+
+# Existing platforms
+
+We will now discuss a few platforms that can be used to set up a blockchain and also compare Bluemix and Azure.
+
+The first one is [__Ethereum__](https://www.ethereum.org/), a public blockchain. Ethereum looks like the Bitcoin blockchain, but it uses Ether as the currency.
+It is faster than Bitcoin with a transaction taking 7 seconds instead of 10 minutes.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Seven instead of 7
Ten instead of 10

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 24]
```diff
@@ -0,0 +1,422 @@
...
+  <div class="col-md-offset-3 col-md-6" style="padding: 0;">
+	{% include image.html img="/img/blockchain/partners.png" alt="Companies and consortiums" title="Companies and consortiums" caption="Table: IBM is part of Hyperledger and Microsoft is part of Enterprise Ethereum Alliance." %}
+</div>
+</div>
+
+
+# Thinking decentralized
+
+Last year, Ken Coenen gave a presentation about the popularity of APIs, and how companies team up to create innovative solutions.
+Data is freed from their silo's and made available through APIs.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Silos

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 25]
```diff
@@ -0,0 +1,422 @@
...
+It's consumable for other departments and even other companies.
+However, when you think of it, all of this data is centralized and we need extra effort to expose it to other parties.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: ... think **about** it ...

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 26]
```diff
@@ -0,0 +1,422 @@
...
+
+When working with blockchain technology, your data is __decentralized by nature__.
+It's funny when you think of it...
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: think **about** it

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 27]
```diff
@@ -0,0 +1,422 @@
...
+We'll give an example.
+All applications implement their own user profile functionality.
+All of this user data - __your profile information__ - is duplicated across many companies.
+It's already a big improvement that applications allow you to use another platform's credentials.
+Logging in with your Facebook or Google account is becoming a habit.
+This gives the end user a way to minimize his/her digital footprint.
+Don Tapscott explains this really well in summer 2016's TED Talk [How the blockchain is changing money and business](https://youtu.be/og7PCYlDYsU?t=849).
+Of course, blockchain technology is still in its early stages.
+It's not even sure whether the technology will last.
+Although these statements are purely hypothetical, I find much food for thought in them.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: We instead of I?

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 28]
```diff
@@ -0,0 +1,422 @@
...
+When talking with people about the possibilities of blockchain, it quickly becomes clear that we still have a long way to go.
+People aren't waiting for yet another technological revolution.
+Instead, we need to start small.
+Blockchain and distributed ledger technology in general will have to evolve naturally.
+
+Blockchain solutions like IBM Blockchain or Microsoft Azure Blockchain-as-a-Service make the technology very accessible to companies in an early stage.
+We believe that a __private blockchain network__ is the __best way to start__ for a company because of the following reasons:
+
+* Throwing all your data at the world is still a very scary idea
+* You have to take all __legal aspects__ into account (think of the EU's new General Data Protection Regulation)
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Perhaps make the EU's regulation clickable and have it link to the actual regulation (if something like it exists)?
I assume there are plenty of people that are unaware of it.

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 29]
```diff
@@ -0,0 +1,422 @@
...
+* You can start small and expose some transactions by defining permissions
+
+Companies are starting to develop applications on their proprietary Bluemix- or Azure platform, without exposing everything to the outside world.
+Get inspired by visiting [State of the DApps](http://dapps.ethercasts.com/).
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Ðapps instead of DApps?

**Comment by kencoenen** [comment]
@kevinnio1 @JeroenDePrest please process @yannickdeturck's feedback. Thanks !

File: _posts/2017-05-07-Blockchain-Introduction.md (hunk 3 above)
Comment (JeroenDePrest) [inline _posts/2017-05-07-Blockchain-Introduction.md]: This is a quote from a book so I will leave it like this.

Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Good point. :-)

**Comment by JeroenDePrest** [comment]
The changes from @yannickdeturck  have been processed. @kencoenen @kevinnio1

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 30]
```diff
@@ -0,0 +1,422 @@
...
+* and a __merkle root hash__
+
+A __merkle root__ does not verify transactions, but verifies a set of transactions.
+Transaction IDs are hashes of the transaction, and the merkle tree is constructed from these hashes.
+It means that if a single detail in any of the transactions changes, so does the merkle root. 
+It also means that if the exact same transactions are listed in a different order, the merkle root will also change.
+So the merkle root is cryptographic proof of the transactions in the block together with the order they are in.
+
+The __nonce number__ specifies that the block root hash must start with a specified number, like four leading zeros.
+This way**,** blockchain is very secure.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: I think the asterixes were copy-pasted?
I only included them in the review to mark the comma in bold.

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 31]
```diff
@@ -0,0 +1,422 @@
...
+So what is it?  
+Smart contracts are traditional contracts and official documents, but written in code.
+As such, the contract is understandable for everyone across the globe, irrespective of the jurisdiction it is related to. 
+Smart contracts are like **If This Then That** statements, only they tend to be a lot more complex.
+
+The different definitions usually fall into one of the following two categories:
+
+1. Sometimes the term is used to identify a specific technology.
+__Pieces of code__ that are stored, verified and executed on a blockchain.
+For example**,** a hello world program.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Remove the asterixes here as well.

File: _posts/2017-05-07-Blockchain-Introduction.md [hunk 32]
```diff
@@ -0,0 +1,422 @@
...
+4. [Permanence](#permanence)
+5. [Security](#security)
+6. [Decentralized](#thinking-decentralized)
+
+We'll discuss each element in detail and explain why blockchain technology can be an answer to that business problem.
+
+
+### Transparency
+
+In a public blockchain network**,** by default every member of the ecosystem can access all transactions stored in the chain.
```
Comment (yannickdeturck) [inline _posts/2017-05-07-Blockchain-Introduction.md]: Remove the astrixes.

**Comment by JeroenDePrest** [comment]
The comma problem has been fixed @yannickdeturck

**Comment by yannickdeturck** [comment]
Perhaps the sole thing to do is renaming the blogpost file to set the correct publication timestamp.
`git mv _posts/2017-05-07-Blockchain-Introduction.md _posts/2017-05-10-Blockchain-Introduction.md`

**Comment by JeroenDePrest** [comment]
Date should be changed now @yannickdeturck @kencoenen

**Comment by yannickdeturck** [comment]
Good job, guys!
Now awaiting the green light of our blockchain expert, @kencoenen before merging.

**Comment by yannickdeturck** [comment]
Green light has been given by @kencoenen , merging...
